modal/_container_entrypoint.py

```diff
@@ -752,6 +754,12 @@ def main(container_args: api_pb2.ContainerArguments, client: Client):
             app: App = synchronizer._translate_out(active_app, Interface.BLOCKING)
             app._init_container(client, container_app)
 
+        # Patch socket.close() to prevent lingering open connections due to TIME_WAIT.
+        # Refer to modal._contaienr_utils.patch_socket_close() for full extent of patch.
+        if is_checkpointing_function:
+            logger.debug("patching socket.close() for memory snapshot")
+            patch_socket_close()
```

Comment:
We need to patch after imports but before running any lifecycle methods, that's why we can't combine them. Patching needs to be done after imports to prevent new imports overriding the patch.